main.go

```diff
@@ -88,108 +96,175 @@ func (c *DownloadPlugin) Run(cliConnection plugin.CliConnection, args []string)
 		os.Exit(1)
 	}
 
-	flagVals := ParseFlags(args)
+	cmdExec := cmd_exec.NewCmdExec()
+
+	// parse flags and get list of download paths
+	flagVals, paths := ParseArgs(args)
+	paths = ExpandGlobs(cmdExec, paths, flagVals.Instance_flag)
 
-	// flag variables
-	filterList := filter.GetFilterList(flagVals.Omit_flag, flagVals.Verbose_flag) // get list of things to not download
+	// get list of things to not download
+	filterList := filter.GetFilterList(flagVals.Omit_flag, flagVals.Verbose_flag)
 
+	// get server path to download from and local path to download to
 	workingDir, err := os.Getwd()
 	check(err, "Called by: Getwd")
-	rootWorkingDirectoryServer, startingPathServer := GetDirectoryContext(workingDir, args)
-	rootWorkingDirectoryLocal := filepath.FromSlash(rootWorkingDirectoryServer)
+	pathVals := GetDirectoryContext(workingDir, paths, flagVals.File_flag)
+	for _, v := range pathVals {
+		v.RootWorkingDirectoryLocal = filepath.FromSlash(v.RootWorkingDirectoryLocal)
+	}
 
 	// ensure cf_trace is disabled, otherwise parsing breaks
 	if os.Getenv("CF_TRACE") == "true" {
 		fmt.Println("\nError: environment variable CF_TRACE is set to true. This prevents download from succeeding.")
 		return
 	}
 
-	// prevent overwriting files
-	if Exists(rootWorkingDirectoryLocal) && flagVals.OverWrite_flag == false {
-		fmt.Println("\nError: destination path", rootWorkingDirectoryLocal, "already Exists and is not an empty directory.\n\nDelete it or use 'cf download APP_NAME --overwrite'")
-		os.Exit(1)
-	}
+	// download files at each input path
+	for _, v := range pathVals {
+		// prevent overwriting files
+		if Exists(v.RootWorkingDirectoryLocal) && flagVals.OverWrite_flag == false {
+			fmt.Println("\nError: destination path", v.RootWorkingDirectoryLocal, "already exists.\n\nDelete it or rerun the command with the '--overwrite' flag.")
+			os.Exit(1)
+		}
 
-	cmdExec := cmd_exec.NewCmdExec()
-	parser = dir_parser.NewParser(cmdExec, appName, flagVals.Instance_flag, onWindows, flagVals.Verbose_flag)
-	dloader = downloader.NewDownloader(cmdExec, &wg, appName, flagVals.Instance_flag, rootWorkingDirectoryServer, flagVals.Verbose_flag, onWindows)
+		// remove files to be overwritten
+		if flagVals.OverWrite_flag {
+			err := os.RemoveAll(v.RootWorkingDirectoryLocal)
+			check(err, "Cannot remove "+v.RootWorkingDirectoryLocal+" for overwrite.")
+		}
 
-	// parse the directory
-	files, dirs := parser.ExecParseDir(startingPathServer)
+		parser = dir_parser.NewParser(cmdExec, appName, flagVals.Instance_flag, onWindows, flagVals.Verbose_flag)
+		dloader = downloader.NewDownloader(cmdExec, &wg, appName, flagVals.Instance_flag, v.RootWorkingDirectoryServer, flagVals.Verbose_flag, onWindows)
```

Comment:
This happens when `wg.Add(1)` is called before the call to `Download` on line 154.